lib/galaxy/tools/verify/__init__.py

```diff
@@ -103,8 +103,9 @@ def verify(
         try:
             compare = attributes.get('compare', 'diff')
             if attributes.get('ftype', None) in ['bam', 'qname_sorted.bam', 'qname_input_sorted.bam', 'unsorted.bam']:
-                local_fh, temp_name = _bam_to_sam(local_name, temp_name)
-                local_name = local_fh.name
+                if attributes.get("decompress", None):
```

Comment:
Shouldn't the default be True here?

Reply:
Good question!

I copied it from here: https://github.com/galaxyproject/galaxy/blob/c1a6a2a48f4c8bdf6682b8ab842a0cd391d40ca9/lib/galaxy/tools/verify/__init__.py#L187 and my intuition was mislead (I guessed that there None means do not decompress, but it does). 

The default value of the decompress attribute seems to be not explicitly documented in the tool xsd file .. but I would interpret it as `False`. As far as I can see also the parser seems to get a default of `False`: https://github.com/galaxyproject/galaxy/blob/2376d7f2fd57fcf9ae00264c4244ba06d63b8906/lib/galaxy/tools/parser/xml.py#L523

So we should make this consistent here .. and add more explicit documentation. I think a default of True should work in most cases and seems to be convenient for tool developers. 

What do you think?

Reply:
So the problem I guess is that ``decompress`` is ``False`` by default in general but this bam->sam behavior is ``True`` by default. I'm not necessarily eager to change either default without a profile version flag and who wants to add that complexity...

Can we catch the exception in the sam conversion and just compare them as bams if that fails? I guess that has issues since the semantics of the other attributes changes a bit.

Reply:
I hope my commit is what you meant.